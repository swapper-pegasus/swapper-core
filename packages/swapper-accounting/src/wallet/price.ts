/* eslint-disable import/prefer-default-export */
import {
  SwapperCoingeckoClient,
  ResponseGetTokenPrice,
  ResponseGetTokens
} from '@swapper-org/swapper-coingecko-client'
import { SUPPORTED_TOKENS } from '../constants'

const coingeckoClient: SwapperCoingeckoClient = new SwapperCoingeckoClient(
  process.env.COINGECKO_ENDPOINT || ''
)

function getSupportedTokenPriceInUsd (): Promise<ResponseGetTokenPrice> {
  return coingeckoClient.getTokensPriceInUsd({ tokens: SUPPORTED_TOKENS })
}

function getTokensDataBySymbol (): Promise<ResponseGetTokens> {
  return coingeckoClient.getTokenDataBySymbol(SUPPORTED_TOKENS)
}

export { getSupportedTokenPriceInUsd, getTokensDataBySymbol }
