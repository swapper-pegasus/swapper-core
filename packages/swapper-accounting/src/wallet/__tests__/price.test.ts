import { getSupportedTokenPriceInUsd, getTokensDataBySymbol } from '../price'

describe('Price', () => {
  it('getSupportedTokenPriceInUsd', async () => {
    const result = await getSupportedTokenPriceInUsd()
    expect(result).toMatchSnapshot()
  })
  it('getTokensDataBySymbol', async () => {
    const result = await getTokensDataBySymbol()
    expect(result).toMatchSnapshot()
  })
})
