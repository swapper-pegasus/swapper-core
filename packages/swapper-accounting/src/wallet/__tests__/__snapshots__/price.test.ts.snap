// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price getSupportedTokenPriceInUsd 1`] = `
Object {
  "bitcoin": Object {
    "usd": "100000",
  },
  "ethereum": Object {
    "usd": "1000",
  },
}
`;

exports[`Price getTokensDataBySymbol 1`] = `
Array [
  Object {
    "id": "ethereum",
    "name": "Ethereum",
    "symbol": "eth",
  },
  Object {
    "id": "bitcoin",
    "name": "Bitcoin",
    "symbol": "btc",
  },
]
`;
